akka-docs/src/test/java/jdocs/circuitbreaker/DangerousJavaActor.java

```diff
@@ -24,13 +24,7 @@ public class DangerousJavaActor extends AbstractActor {
   private final LoggingAdapter log = Logging.getLogger(getContext().system(), this);
 
   public DangerousJavaActor() {
-    this.breaker =
-        new CircuitBreaker(
-                getContext().getDispatcher(),
-                getContext().getSystem().getScheduler(),
-                5,
-                Duration.ofSeconds(10),
-                Duration.ofMinutes(1))
+    this.breaker = CircuitBreaker.lookup("dangerous-breaker", getContext().getSystem())
```

Comment:
This is actually not all that great, because the callback added below calls a local method in the actor, setting us up for doing the wrong thing touching actor state on another thread, and also a memory leak from the actor stopping and leaving that callback in the breaker so that the actor is kept in memory even after stopping.

Reply:
I agree, would be good to rewrite the example, and use Behavior and pipeToSelf. We have a `DataAccess` example (used in stash docs) that could be a good example of external interaction from actor where a circuit breaker is useful. If using that it should be copied for this purpose to not make the stashing docs more involved.

Reply:
I looked a bit at copying that but I don't think that's great either, it crashes the actor on failure, so the circuit breaker isn't quite shown as useful then. I'll have to adapt quite a bit, but maybe that is fine.

Reply:
Gave it a shot, just the Scala side until I get some feedback